### PR TITLE
Add e2e test for cluster claim

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-2143

### CODE REVIEW

**Comment by hongkailiu** [comment]
Nice. Reached the expected failing point.
```
{"level":"error","msg":"  * could not run steps: step cluster-pool:e2e-claim:ocp-4.7-amd64-aws-fake failed: failed to find a cluster pool providing the cluster: map[architecture:amd64 cloud:aws owner:fake product:ocp version:4.7]","time":"2021-04-20T20:07:36Z"}
```

File: test/e2e/multi-stage/e2e_test.go [hunk 1]
```diff
@@ -121,12 +121,22 @@ func TestMultiStage(t *testing.T) {
...
+			success: true,
+			output:  []string{"TODO"},
+		},
 	}
 
 	for _, testCase := range testCases {
 		testCase := testCase
 		framework.Run(t, testCase.name, func(t *framework.T, cmd *framework.CiOperatorCommand) {
 			cmd.AddArgs(testCase.args...)
+			if testCase.name == "e2e-claim" {
```
Comment (stevekuznetsov) [inline test/e2e/multi-stage/e2e_test.go:137]: No. Add this as a field to the test case or something - keying off of a string name is fragile.

**Comment by stevekuznetsov** [comment]
Part of DPTP-2099

File: test/e2e/multi-stage/cluster-claim.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+base_images:
+  cli:
+    name: "4.8"
+    namespace: ocp
+    tag: cli
+  os:
```
Comment (stevekuznetsov) [inline test/e2e/multi-stage/cluster-claim.yaml:6]: What is this image used for?

File: test/e2e/multi-stage/cluster-claim.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      architecture: amd64
+      cloud: aws
+      owner: fake
+      product: ocp
+      timeout: 1h0m0s
+      version: "4.7"
+    steps:
+      test:
+        - as: claim-step
+          commands: |
```
Comment (stevekuznetsov) [inline test/e2e/multi-stage/cluster-claim.yaml:28]: Which of these commands will fail if the claim does not work? Why are all of the commands necessary?

File: test/e2e/multi-stage/e2e_test.go [hunk 4]
```diff
@@ -121,12 +122,22 @@ func TestMultiStage(t *testing.T) {
 			success: true,
 			output:  []string{`run-as-script-success succeeded`},
 		},
+		{
+			name:    "e2e-claim",
+			args:    []string{"--unresolved-config=cluster-claim.yaml", "--target=e2e-claim"},
+			env:     []string{defaultJobSpec},
+			success: true,
+			output:  []string{"TODO"},
```
Comment (stevekuznetsov) [inline test/e2e/multi-stage/e2e_test.go:130]: ?

File: test/e2e/multi-stage/e2e_test.go [hunk 5]
```diff
@@ -121,12 +122,22 @@ func TestMultiStage(t *testing.T) {
...
+		},
 	}
 
 	for _, testCase := range testCases {
 		testCase := testCase
 		framework.Run(t, testCase.name, func(t *framework.T, cmd *framework.CiOperatorCommand) {
 			cmd.AddArgs(testCase.args...)
+			if testCase.needHive {
```
Comment (stevekuznetsov) [inline test/e2e/multi-stage/e2e_test.go:138]: I don't see any test case setting this

File: test/e2e/multi-stage/cluster-claim.yaml (hunk 2 above)
Comment (hongkailiu) [reply test/e2e/multi-stage/cluster-claim.yaml]: Removed.

File: test/e2e/multi-stage/cluster-claim.yaml (hunk 3 above)
Comment (hongkailiu) [reply test/e2e/multi-stage/cluster-claim.yaml:24]: I have to see what hive gives us in the secret and just my cmd accordingly.

File: test/e2e/multi-stage/e2e_test.go (hunk 5 above)
Comment (hongkailiu) [reply test/e2e/multi-stage/e2e_test.go:139]: My bad.
Updated.

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**